helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/WagedValidationUtil.java

```diff
@@ -88,4 +90,16 @@
     }
     return partitionCapacity;
   }
+
+  /**
+   * Checks whether or not a resource has enabled WAGED rebalancer.
+   *
+   * @param idealState {@code IdealState} of the resource being checked.
+   * @return {@code true} if WAGED is enabled; otherwise, {@code false}.
+   */
+  public static boolean isWagedEnabled(IdealState idealState) {
```

Comment:
This could be used in `BestPossibleStateCalcStage.computeResourceBestPossibleStateWithWagedRebalancer` as well. Could you check for other similar usages? Replacing those logic with this function is a step up for cleanness.

Reply:
Ideally, replacing those places with this API should be done in another PR. Since you've pointed out that place, I just made the change. If there are more places, let's make the change in another PR.